st2api/st2api/controllers/v1/datastore.py

```diff
@@ -34,15 +34,22 @@ class KeyValuePairController(RestController):
     """
 
     @jsexpose(str)
-    def get_one(self, id):
+    def get_one(self, name):
```

Comment:
In my mind this starts the entire is name unique discussion. The minute we allow user scoped keys name will not remain unique and this API will fall apart. I am not positive we should make this a get by name API.